File: tests/fixtures/real_world/javascript/taint/sqli_concat.js
var mysql = require('mysql');

function getUser(connection, userId) {
    var query = 'SELECT * FROM users WHERE id = ' + userId;
    connection.query(query, function(err, results) {
        return results;
    });
}

function getUserSafe(connection, userId) {
    connection.query('SELECT * FROM users WHERE id = ?', [userId], function(err, results) {
        return results;
    });
}

File: tests/fixtures/real_world/python/taint/sqli_concat.py
from flask import Flask, request
import sqlite3

app = Flask(__name__)

@app.route('/user')
def get_user():
    user_id = request.args.get('id')
    conn = sqlite3.connect('app.db')
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE id = " + user_id)
    return str(cursor.fetchall())

@app.route('/user-safe')
def get_user_safe():
    user_id = request.args.get('id')
    conn = sqlite3.connect('app.db')
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE id = ?", (user_id,))
    return str(cursor.fetchall())
